_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml

```diff
@@ -320,6 +342,409 @@
         <CombineArray value="AP_IsVanillaCommandCenterFlying"/>
         <CombineArray value="AP_IsAPCommandCenterFlying"/>
     </CValidatorCombine>
+    <CValidatorCombine id="AP_IsInvulnerableAndHasSOAHeroicShieldImmunity">
+        <Type value="And"/>
+        <CombineArray value="AP_IsInvulnerable"/>
+        <CombineArray value="AP_HasSOAHeroicShieldImmunity"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsSOACaster">
+        <Value value="AP_SoACaster"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaSOACaster">
+        <Value value="SoACaster"/>
+    </CValidatorUnitType>
+    <CValidatorUnitFilters id="AP_IsInvulnerable">
+        <Filters value="Invulnerable;-"/>
+    </CValidatorUnitFilters>
+    <CValidatorUnitType id="AP_IsAPHerculesCombine"/>
+    <CValidatorCombine id="AP_IsVanillaHerculesCombine">
+        <CombineArray value="AP_IsVanillaHercules"/>
+        <CombineArray value="AP_IsVanillaHerculesLander"/>
+        <CombineArray value="AP_IsVanillaHerculesLanderFlying"/>
+    </CValidatorCombine>
+    <CValidatorCombine id="IsHerculesCombine" parent="CValidatorCombine">
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaHerculesCombine"/>
+        <CombineArray value="AP_IsAPHerculesCombine"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsAPHercules">
+        <Value value="AP_Hercules"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaHercules">
+        <Value value="Hercules"/>
+    </CValidatorUnitType>
+    <CValidatorCombine id="IsHercules" parent="CValidatorCombine">
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaHercules"/>
+        <CombineArray value="AP_IsAPHercules"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsAPHerculesLander">
+        <Value value="AP_HerculesLander"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaHerculesLander">
+        <Value value="HerculesLander"/>
+    </CValidatorUnitType>
+    <CValidatorCombine id="IsHerculesLander" parent="CValidatorCombine">
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaHerculesLander"/>
+        <CombineArray value="AP_IsAPHerculesLander"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsAPHerculesLanderFlying">
+        <Value value="AP_HerculesLanderFlying"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaHerculesLanderFlying">
+        <Value value="HerculesLanderFlying"/>
+    </CValidatorUnitType>
+    <CValidatorCombine id="IsHerculesLanderFlying" parent="CValidatorCombine">
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaHerculesLanderFlying"/>
+        <CombineArray value="AP_IsAPHerculesLanderFlying"/>
+    </CValidatorCombine>

```

Comment:
Again, check as above

Reply:
Looks like the others are too affected. Some are correct, like for Archon, but some are not